app/middlewares/auth/token.js

```diff
@@ -23,3 +25,29 @@ exports.validateToken = (req, res, next) => {
     return next(invalidToken(TOKEN_INVALID));
   }
 };
+
+exports.validateTokenAdmin = async (req, res, next) => {
+  try {
+    const authorization = req.header('authorization');
+
+    if (!authorization) return next(invalidToken(TOKEN_REQUIRED));
+
+    const [, token] = authorization.split(' ');
+
+    const user = decoded(token);
+
```

Comment:
To avoid rewriting the same code from the validateToken middleware, create one that just checks if the user is in role